src/ng/browser.js

```diff
@@ -60,6 +73,33 @@ function Browser(window, document, $log, $sniffer) {
     }
   }
 
+  function decOutstandingRequestCount(taskType) {
+    taskType = taskType || DEFAULT_TASK_TYPE;
+    if (outstandingRequestCounts[taskType]) outstandingRequestCounts[taskType]--;
+    if (outstandingRequestCounts[ALL_TASKS_TYPE]) outstandingRequestCounts[ALL_TASKS_TYPE]--;
```

Comment:
I feel like this should go into the `if (outstandingRequestCounts[taskType])` and not be it's own `if`. That way if `taskType` is invalid or not set it won't decrement `outstandingRequestCounts[ALL_TASKS_TYPE]`...?